src/main/java/com/gluonhq/substrate/target/AbstractTargetConfiguration.java

```diff
@@ -244,4 +254,34 @@ public boolean runUntilEnd(Path appPath, String appName) throws IOException, Int
         return true;
     }
 
+    // Default settings below, can be overridden by subclasses
+
+    String getAdditionalSourceFileLocation() {
+        return "/native/linux/";
+    }
+
+    List<String> getAdditionalSourceFiles() {
+        return Arrays.asList("launcher.c","thread.c");
+    }
+
+    String getCompiler() {
+        return "gcc";
+    }
+
+    String getLinker() {
+        return "gcc";
+    }
+
+    List<String> getTargetSpecificLinkFlags() {
+        return new LinkedList<>();
```

Comment:
This is not a good representation of empty list. List.of() or Collections.emptyList() are better. They don't do any allocations... Same for below methods

Reply:
Done